pkg/cleaner/cleaning.go

```diff
@@ -105,12 +109,26 @@ func (c *FileContentObfuscator) ObfuscateFile(inputFile string, outputFile strin
 	// we need to assess whether the file exists already to ensure we don't overwrite existing obfuscated data.
 	// that can happen while obfuscating file names and their paths.
 	// Additionally, the stat check is required because os.O_CREATE will implicitly os.O_TRUNC if a file already exist
+	c.pathCollisionMutex.Lock()
+	defer c.pathCollisionMutex.Unlock()
+
 	_, err = os.Stat(writePath)
 	if err != nil && !os.IsNotExist(err) {
 		return fmt.Errorf("failed to determine if %s already exists: %w", writePath, err)
 	}
 	if err == nil {
-		return fmt.Errorf("file %s already exists, check whether the obfuscators overwrite each other", writePath)
+		var fileExt int
```

Comment:
why not just `fileExt:=0`?

Reply:
The behavior is same in either of the cases

Reply:
this is correct, it just looks weird for the reader as there is no need to supply the type. But it's just my personal preference, so feel free to leave it